### PR TITLE
[Feat]: Download as CSV and Execl and Search option for key-operation table

### PR DESCRIPTION
**Pull request by Rajat-Dabade**
Close: https://github.com/SigNoz/signoz/issues/3832
Close: https://github.com/SigNoz/signoz/issues/3833

### CODE REVIEW

**Comment by palashgdev** [comment]
@Rajat-Dabade this method does not take unit into account please fix this one

example: 
Screenshot 2023-10-31 at 3 48 55 PM

**Comment by palashgdev** [comment]
ideally this downloading of csv should be handled from the backend itself

**Comment by ankitnayan** [comment]
> ideally this downloading of csv should be handled from the backend itself taking the account of pagination in future it will be hard for the user/ client to aggregate the data

yeah..when pagination comes to play or there is huge data the search should also query the backend. But this data is not supposed to grow big so we should be good

**Comment by ankitnayan** [comment]
Screenshot 2023-10-31 at 9 21 53 PM

1. Name should be the 1st column
2. P50 should be `P50 (in ms)` and the same for the other P95 and P99
3. P50 (in ms) value should be in `ms`. The current values are in nano. 
4. Change styling after UX consultation in a separate PR @GeekBoySupreme

**Comment by ankitnayan** [comment]
- It's `Error (%)` and not `Error Count` and the value is also incorrect
- the values of `P50`, `P95` and `P99` are duplicated and incorrect. Please verify data sanity before pushing any changes
- append name of the service to filename? `top-operations-customer.xls/csv`

**Review by ankitnayan — DISMISSED** [review]

**Comment by ankitnayan** [comment]
@Rajat-Dabade can you also check why the numbers are treated as text in `xls` but as numbers in `csv`. We should treat them as numbers. Selecting multiple values in the column `P50` should show Sum for numbers.

**Review by palashgdev — APPROVED** [review]

File: frontend/src/container/QueryTable/utils.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { RowData } from 'lib/query/createTableColumnsFromQuery';
+
+export function createDownloadableData(
+	inputData: RowData[],
+): Record<string, string>[] {
+	return inputData.map((row) => ({
```
Comment (YounixM) [inline frontend/src/container/QueryTable/utils.ts:6]: Add safety checks wherever required. We should handle safety checks everywhere by default when not strictly typed.

Comment (Rajat-Dabade) [reply frontend/src/container/QueryTable/utils.ts:6]: Added the safety check.

**Review by YounixM — APPROVED** [review]

**Pull request merged**